docs/architecture/pairing-with-use-integration-system.md

```diff
@@ -0,0 +1,86 @@
+
+Mutations `generateAuthForApplication(id: ID!, type: PairingAdapterAuthTypeInput!, inputParams: JSON!)` abstracts the way of obtaining trusted relation data from the Integration System. Calling the mutation with appropriate parameters checks whether the desired Application is under control of the Integrations System. If the underlying Integration System has Pairing Adapter assigned, the call is enriched with additional data defined for Integration System and data which comes from the request context (eg. user ID). Next, the request is handed to the Pairing Adapter. The Pairing Adapter implements the logic required for obtaining all required data from the Integration System Instance based on the incoming request.
+
+In case the Application is not related to any Integration System the `generateAuthForApplication(id: ID!, type: PairingAdapterAuthTypeInput!, inputParams: JSON!)` mutation abstracts fetching the one-time token from the Connector or client credentials, according to the credential type described by the `type` parameter.
+
+The Integration System can support multiple auth types. To determine what are the supported types for a particular Application, the API offers appropriate query: `getAuthTypesForApplication(id: ID!)`. The response for the query returns the possible values for the `type` parameter for auth generation mutation.
+
+## Pairing Adapter
+
+Pairing Adapter is a component that runs in the Compass environment. Its API is not exposed outside the cluster. It is responsible for handling specific logic to obtain auth details required to establish a trusted relation between the Application Instance and Integration System Instance. Each Pairing Adapter is coupled with one Integration System.
+
+### Pairing Adapter contract
+
+The component to be able to act as a Pairing Adapter must fulfill the following contract. 
+
+>**NOTE** The component offers the GraphQL API. All the queries and mutations work in a synchronous manner.
+
+1. It offers the query `getAuthTypes(): [PairingAdapterAuthType!]` that returns collection of `PairingAdapterAuthType`.
+
+1. It offers the mutation `generateAuth(type: PairingAdapterAuthType, in: PairingAdapterParamsInput): PairingAuthResult!` that handles the process of obtaining the pairing credentials from the Integration System Instance.
+
+`PairingAdapterAuthType` is the type that describes supported auth types for the PairingAdapter.
+
+`PairingAdapterParams` is the type that describes the payload which is passed from the Director component to the Pairing Adapter. This payload should be sufficient for Pairing Adapter to obtain pairing data from the Integration System Instance. The response from this mutation returns `PairingAuthResult` type.
+
+```graphql
+input PairingAdapterAuthTypeInput {
```

Comment:
If this would be an enum, then you need to modify the API every time you add a new PairingAdapter with new Auth type to support it on the Directors API level.